### PR TITLE
[SYCL][ESIMD] Add new test for simd::copy_to/from with alignment flags

### PR DESCRIPTION
**Pull request by sndmitriev**
Signed-off-by: Sergey Dmitriev <serguei.n.dmitriev@intel.com>

### CODE REVIEW

**Comment by sndmitriev** [comment]
I believe sycl-ls fail in precommit testing on windows is unrelated to my patch.

[2021-11-17T04:34:38.158Z] Failed Tests (1):
[2021-11-17T04:34:38.158Z]   SYCL :: Plugin/sycl-ls.cpp

File: SYCL/ESIMD/api/simd_copy_to_from.cpp [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+    Pass &= testUSM<T, 31>(Type, Q);
+    Pass &= testUSM<T, 32>(Type, Q);
+  }
+
+  return Pass;
+}
+
+template <typename T, int N> bool testAcc(const std::string &Type, queue &Q) {
+  std::unique_ptr<T> Src(new T[512u]);
```
Comment (kbobrovs) [inline SYCL/ESIMD/api/simd_copy_to_from.cpp]: strictly speaking, alignment of `new` result is not known, so either aligned allocation should be used or buffer start should be enforced to known alignment ((ptr + align) & ~align)

Comment (sndmitriev) [reply SYCL/ESIMD/api/simd_copy_to_from.cpp]: For SVM test for sure. But is it required for the buffer + accessor as well?

Comment (kbobrovs) [reply SYCL/ESIMD/api/simd_copy_to_from.cpp]: accessor is wrapped around the pointer allocated by `new` in the test, so I think yes, it is.
For buffer w/o host pointer it is likely that allocated memory would be page-aligned, but I'm not sure if this is guaranteed or not, so even in that case forcing alignment would make sense (esp. overaligned).

Comment (sndmitriev) [reply SYCL/ESIMD/api/simd_copy_to_from.cpp]: Okay, if that is the case I will change this to aligned allocation.

**Review by kbobrovs — APPROVED** [review]

**Pull request merged**